service/src/main/java/bio/terra/cbas/controllers/ControllerUtils.java

```diff
@@ -0,0 +1,31 @@
+package bio.terra.cbas.controllers;
+
+import bio.terra.cbas.dependencies.sam.SamService;
+import bio.terra.cbas.util.BearerTokenFilter;
+import java.util.Optional;
+import org.broadinstitute.dsde.workbench.client.sam.model.UserStatusInfo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.context.request.RequestContextHolder;
+
+@Component
```

Comment:
For you to think about, my inclination would be to remove the `Component`-ness from this and make this a more pure utility class (ie, no instances, all static functions, each function gets the values it needs to do its job as arguments).

Right now the only reason for this to be a Component is to get a `SamService`, but the callers are more naturally in the "component" space so they should be able to autowire in a SamService in a way that I _think_ would feel more natural.